src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -65,6 +69,7 @@ private SwitchBootstraps() {}
 
     private static final Object SENTINEL = new Object();
     private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+    private static final boolean previewEnabled = true;
```

Comment:
not sure about the use of this constant, do we really need it?

Reply:
Uh, I think this is a good catch.

I am not completely sure what the policy is, but here we are enhancing a non-preview method with some preview-based behavior. I would feel better if the method would refuse to work in the preview way when preview is not enabled. I.e. my opinion is that this should either be initialized to `jdk.internal.misc.PreviewFeatures.isEnabled()`, or `jdk.internal.misc.PreviewFeatures.ensureEnabled()` should be used appropriately.

Reply:
Use the `jdk.internal.misc.PreviewFeatures.isEnabled()` and remove an erroneous check in `typeSwitch`.

Reply:
Addressed in https://github.com/openjdk/jdk/pull/15638/commits/663c7a54cec9ffd6ca468080f4d084e5fa4c9c3d